packages/web-app-files/src/store/actions.js

```diff
@@ -36,6 +38,76 @@ export default {
       context.commit('ADD_FILE_SELECTION', file)
     }
   },
+  copySelectedFiles(context) {
+    context.commit('CLIPBOARD_SELECTED')
+    context.commit('SET_CLIPBOARD_ACTION', ClipboardActions.Copy)
+    context.dispatch(
+      'showMessage',
+      {
+        title: $gettext('Copied to clipboard!'),
+        status: 'success'
+      },
+      { root: true }
+    )
+  },
+  cutSelectedFiles(context) {
+    context.commit('CLIPBOARD_SELECTED')
+    context.commit('SET_CLIPBOARD_ACTION', ClipboardActions.Cut)
+    context.dispatch(
+      'showMessage',
+      {
+        title: $gettext('Copied to clipboard!'),
+        status: 'success'
+      },
+      { root: true }
+    )
+  },
+  async pasteSelectedFiles(
+    context,
+    {
+      client,
+      createModal,
+      hideModal,
+      showMessage,
+      $gettext,
+      $gettextInterpolate,
+      $ngettext,
+      upsertResource
+    }
+  ) {
+    let movedResources
+    if (context.state.clipboardAction === ClipboardActions.Cut) {
+      movedResources = await move(
+        context.state.clipboardResources,
+        context.state.currentFolder,
+        client,
+        createModal,
+        hideModal,
+        showMessage,
+        $gettext,
+        $gettextInterpolate,
+        $ngettext
+      )
+      context.commit('CLEAR_CLIPBOARD')
+    }
+    if (context.state.clipboardAction === ClipboardActions.Copy) {
+      movedResources = await copy(
+        context.state.clipboardResources,
+        context.state.currentFolder,
+        client,
+        createModal,
+        hideModal,
+        showMessage,
+        $gettext,
+        $gettextInterpolate,
+        $ngettext
+      )
+    }
+    for (const resource of movedResources) {
+      const loadedResource = await client.files.fileInfo(resource.webDavPath, DavProperties.Default)
```

Comment:
is it needed that the op runs one after the other or can it run simultaneous `Promise.all([...])`?